source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -10,78 +10,653 @@ POJO Customization
+- :java-docs:`ClassModelBuilder.conventions() </apidocs/bson/org/bson/codecs/pojo/ClassModelBuilder.html#conventions(java.util.List)>`
+
+To create a custom Convention, create a class that implements the
+``Convention`` interface and override the ``apply()`` method from which you can
+access your ``ClassModelBuilder`` instance.
+
+.. _annotations:
 
 Annotations
 -----------
 
-- Description
-- Table of Annotations
-- Example class with annotations
-- Example output
+You can apply annotations to the getter and setter methods of a POJO
+class. These annotations configure the ``ClassModel`` and ``PropertyModel``
+behavior for a specific field, method, or class.
+
+The following annotations are available from the
+:java-docs:`org.bson.codecs.pojo.annotations </apidocs/bson/org/bson/codecs/pojo/annotations/package-summary.html>`
+package:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Annotation Name
+     - Description
+
+   * - ``BsonCreator``
+     - Marks a public constructor or a public static method as the creator for
+       new instances of the class. You can combine this with the
+       ``BsonProperty`` annotation to link the parameters with properties.
+
+   * - ``BsonDiscriminator``
+     - Specifies a class uses a discriminator. You can set a custom
+       discriminator key and value.
+
+   * - ``BsonRepresentation``
+     - Specify a BSON type to store when different from the POJO property.
+
+   * - ``BsonId``
+     - Marks a property to serialized as the _id property.
+
+   * - ``BsonIgnore``
+     - Marks a property to ignore. You can configure whether to serialized
+       and/or deserialize a property.
+
+   * - ``BsonProperty``
+     - Allows for an alternative document key name when converting the POJO
+       field to BSON. Also allows a field to turn on using a discriminator
+       when storing a POJO value.
+
+The following code snippet shows a sample POJO called ``Product`` that uses
+the annotations described above.
+
+.. _bson-annotation-code-example:
+
+.. code-block:: java
+   :emphasize-lines: 9,11,14,15,18,21
+
+   import org.bson.BsonType;
+   import org.bson.codecs.pojo.annotations.BsonCreator;
+   import org.bson.codecs.pojo.annotations.BsonDiscriminator;
+   import org.bson.codecs.pojo.annotations.BsonId;
+   import org.bson.codecs.pojo.annotations.BsonIgnore;
+   import org.bson.codecs.pojo.annotations.BsonProperty;
+   import org.bson.codecs.pojo.annotations.BsonRepresentation;
+
+   @BsonDiscriminator(value="AnnotatedProduct", key="_cls")
+   public class Product {
+       @BsonProperty("modelName")
+       private String name;
+
+       @BsonId()
+       @BsonRepresentation(BsonType.OBJECT_ID)
+       private String serialNumber;
+
+       @BsonIgnore
+       private List<Product> relatedItems;
+
+       @BsonCreator
+       public ProductPojo(@BsonProperty("modelName") String name) {
```

Comment:
Big fan of this example. One thing that I notice you don't mention is the usage of this ``@BsonProperty`` annotation in the constructor parameters -- should we add this to the BsonProperty bullet point?

Reply:
Since it can't be used by itself in this context, I described under the BsonCreator item https://github.com/mongodb/docs-java/pull/52/files#diff-81770bd4e69d34b6329eacade6c66cffd45edab4a1eb84310dc644364e1660deR261-R264

I think it might be a little confusing or extraneous to talk about how you need to use it only if you decide to use a custom constructor which requires @BsonCreator, and @BsonCreator already mentions it.